### PR TITLE
285 polygon issue with historical option data

### PR DESCRIPTION
**Pull request by davidlatte**
Refactored get_price_from_polygon() completely to be more efficient and eliminate multiple calls from looping. Polygon queries are not padded by 1 day for start/end to eliminate the possibility of intraday gaps in data.
image

### CODE REVIEW

File: tests/test_polygon_helper.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+from lumibot.entities import Asset
+from lumibot.tools import polygon_helper as ph
+
+
+class FakeContract:
+    def __init__(self, ticker):
+        self.ticker = ticker
+
+
+class TestPolygonHelpers:
```
[the remark this answers is not in the document]
Comment (davidlatte) [reply tests/test_polygon_helper.py:17]: @korbit-ai-mentor these test cases are not dependent on each other and can be run in any order.

[the remark this answers is not in the document]
Comment (davidlatte) [reply tests/test_polygon_helper.py:17]: @korbit-ai-mentor these test cases do not have any common setup actions.

File: lumibot/tools/polygon_helper.py [hunk 2]
```diff
@@ -30,233 +23,289 @@ def get_price_data_from_polygon(
...
+    Parameters
+    ----------
+    df_all : pd.DataFrame
+        A DataFrame with the data we already have
+    result : list
+        A List of dictionaries with the new data from Polygon
+        Format: [{'o': 1.0, 'h': 2.0, 'l': 3.0, 'c': 4.0, 'v': 5.0, 't': 116120000000}]
+    """
+    df = pd.DataFrame(result)
+    if df is not None and len(df) > 0:
```
[the remark this answers is not in the document]
Comment (davidlatte) [reply lumibot/tools/polygon_helper.py:283]: @korbit-ai-bit accepted.

File: tests/test_polygon_helper.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidlatte) [reply tests/test_polygon_helper.py:17]: @korbit-ai-mentor the test cases put all persistent data under a pytest tmpdir fixture that automatically gets cleaned up.  There is nothing else to do.

**Review by grzesir — APPROVED** [review]
looks good

**Pull request merged**